src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/curatedcontent/CuratedMainPagePageObject.java

```diff
@@ -0,0 +1,116 @@
+
+/**
+ * @ownership: Content X-Wing
+ */
+public class CuratedMainPagePageObject extends BasePageObject {
+
+  @FindBy(css = ".mobile-top-leaderboard")
+  private WebElement mobileTopLeaderboard;
+  @FindBy(css = ".article-title")
+  private WebElement articleTitle;
+  @FindBy(css = ".wikia-stats-container")
+  private WebElement wikiaStatsContainer;
+  @FindBy(css = ".featured-content")
+  private WebElement featuredContent;
+  @FindBy(css = ".curated-content")
+  private WebElement curatedContent;
+  @FindBy(css = ".mobile-in-content")
+  private WebElement mobileInContent;
+  @FindBy(css = ".trending-articles")
+  private WebElement trendingArticles;
+  @FindBy(css = ".trending-videos")
+  private WebElement trendingVideos;
+  @FindBy(css = ".mobile-prefooter")
+  private WebElement mobilePrefooter;
+
+  @FindBys(@FindBy(css = ".curated-content-item .clamp"))
+  private List<WebElement> itemDisplayNameList;
+
+  public void verifyItem(String itemDisplayName) {
+    verifyTextInListElements(itemDisplayNameList, itemDisplayName);
+  }
+
+  public CuratedContentPageObject clickOnItem(String itemDisplayName) {
+    clickByListElementText(itemDisplayNameList, itemDisplayName);
+    return new CuratedContentPageObject(driver);
+  }
+
+  private enum Settings {
+    TIME_OUT_IN_SEC(5),
+    CHECK_OUT_IN_MILLI_SEC(1000);
+
+    private int value;
+
+    Settings(int value) {
+      this.value = value;
+    }
+  }
+
+  public CuratedMainPagePageObject(WebDriver driver) {
+    super(driver);
+  }
+
+  public int getElementOffsetTop(String element) {
+    JavascriptExecutor js = (JavascriptExecutor) driver;
+    return Integer
+        .parseInt(js.executeScript("return $(arguments[0]).offset().top", element).toString());
+  }
+
+  public boolean isMobileTopLeaderboardVisible() {
+    return isCuratedElementVisible(mobileTopLeaderboard);
+  }
+
+  public boolean isArticleTitleVisible() {
+    return isCuratedElementVisible(articleTitle);
+  }
+
+  public boolean isWikiaStatsContainerVisible() {
+    return isCuratedElementVisible(wikiaStatsContainer);
+  }
+
+  public boolean isFeaturedContentVisible() {
+    return isCuratedElementVisible(featuredContent);
+  }
+
+  public boolean isCuratedContentVisible() {
+    return isCuratedElementVisible(curatedContent);
+  }
+
+  public boolean isMobileInContentVisible() {
+    return isCuratedElementVisible(mobileInContent);
+  }
+
+  public boolean isTrendingArticlesVisible() {
+    return isCuratedElementVisible(trendingArticles);
+  }
+
+  public boolean isTrendingVideosVisible() {
+    return isCuratedElementVisible(trendingVideos);
+  }
+
+  public boolean isMobilePrefooterVisible() {
+    return isCuratedElementVisible(mobilePrefooter);
+  }
+
+  private boolean isCuratedElementVisible(WebElement element) {
+    try {
+      wait.forElementVisible(element, Settings.TIME_OUT_IN_SEC.value,
+                             Settings.CHECK_OUT_IN_MILLI_SEC.value);
+    } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)